[BUG]: SIGSEGV panic in `5.28.0` related to pages feature of github_repository resource

Hi,

I'm observing SIGSEGV panic related to usage of pages in repo resources in provider version in `5.28.0`.

### Terraform Version
Please note that the bug was also observed with terraform version `1.4.6`, but I've later upgraded in an attempt to work around the bug.

```shell
$ terraform version 
Terraform v1.5.0
on linux_amd64
```

### Affected Resource(s)
- github_repository


### Terraform Configuration Files
```hcl
variable "org_name" {
  nullable = false
  type     = string
}

variable "app_id" {
  nullable = false
  type     = string
}

variable "app_installation_id" {
  nullable = false
  type     = string
}

variable "github_auth_app_private_key" {
  nullable  = false
  type      = string
  sensitive = true
}

terraform {
  required_version = ">= 1.4.6, <= 1.5.0"
  required_providers {
    github = {
      source  = "integrations/github"
      version = "~> 5.28.0"
    }
  }
}

provider "github" {
  owner = var.org_name
  app_auth {
    id              = var.app_id
    installation_id = var.app_installation_id
    pem_file        = var.github_auth_app_private_key
  }
}

resource "github_repository" "repo_no_pages" {
  name       = "this-repo-has-no-pages"
  visibility = "public"
}

resource "github_repository" "repo_with_pages" {
  name       = "this-repo-has-pages-enabled"
  visibility = "public"
  auto_init  = true # avoid 'main branch must exist before GitHub Pages can be built'
  pages {
    source {
      branch = "main"
      path   = "/"
    }
  }
  depends_on = [
    github_repository.repo_no_pages
  ]
}
```

### Debug Output
```shell
export TF_LOG="DEBUG"
export TF_LOG_PROVIDER="TRACE"
```
https://gist.github.com/Laffs2k5/0bdeff004b5c6c0fe7e947e327e1b368


### Panic Output

Important part:
```shell
Stack trace from the terraform-provider-github_v5.28.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf9a5a7]

goroutine 115 [running]:
github.com/integrations/terraform-provider-github/v5/github.expandPages({0xc000694e00?, 0x1295cf8?, 0x1b?})
```

For full output from `apply` see https://gist.github.com/Laffs2k5/9abc3ac1e3d6f993689c9fbb83490e6d


### Expected Behavior
Both repos declared should be created. The latter should have the pages feature configured.


### Actual Behavior
Both repos was created. The latter does not have the pages feature configured.


### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Make sure none of the declared repos already exists in the org.
2. Run `terraform init`.
3. Run `terraform apply`.
4. Observe the panic.


### Important Factoids

**Workaround**: To work around this issue downgrade to version `5.27.0` of the provider.

### References
N/A

--- a/github/resource_github_repository.go
+++ b/github/resource_github_repository.go
@@ -836,20 +836,23 @@ func expandPages(input []interface{}) *github.Pages {
 		return nil
 	}
 	pages := input[0].(map[string]interface{})
-	var source *github.PagesSource
-	if pagesSource, ok := pages["source"].([]interface{})[0].(map[string]interface{}); ok {
-		if v, ok := pagesSource["branch"].(string); ok {
-			if v != "" {
+	source := &github.PagesSource{
+		Branch: github.String("main"),
+	}
+	if len(pages["source"].([]interface{})) == 1 {
+		if pagesSource, ok := pages["source"].([]interface{})[0].(map[string]interface{}); ok {
+			if v, ok := pagesSource["branch"].(string); ok {
 				source.Branch = github.String(v)
 			}
-		}
-		if v, ok := pagesSource["path"].(string); ok {
-			// To set to the root directory "/", leave source.Path unset
-			if v != "" && v != "/" {
-				source.Path = github.String(v)
+			if v, ok := pagesSource["path"].(string); ok {
+				// To set to the root directory "/", leave source.Path unset
+				if v != "" && v != "/" {
+					source.Path = github.String(v)
+				}
 			}
 		}
 	}
+
 	var buildType *string
 	if v, ok := pages["build_type"].(string); ok {
 		buildType = github.String(v)

--- a/github/resource_github_repository_test.go
+++ b/github/resource_github_repository_test.go
@@ -859,6 +859,47 @@ func TestAccGithubRepositoryPages(t *testing.T) {
 
 	})
 
+	t.Run("expand Pages configuration with workflow", func(t *testing.T) {
+		input := []interface{}{map[string]interface{}{
+			"build_type": "workflow",
+			"source":     []interface{}{map[string]interface{}{}},
+		}}
+
+		pages := expandPages(input)
+		if pages == nil {
+			t.Fatal("pages is nil")
+		}
+		if pages.GetBuildType() != "workflow" {
+			t.Errorf("got %q; want %q", pages.GetBuildType(), "workflow")
+		}
+		if pages.GetSource().GetBranch() != "main" {
+			t.Errorf("got %q; want %q", pages.GetSource().GetBranch(), "main")
+		}
+	})
+
+	t.Run("expand Pages configuration with source", func(t *testing.T) {
+		input := []interface{}{map[string]interface{}{
+			"build_type": "legacy",
+			"source": []interface{}{map[string]interface{}{
+				"branch": "main",
+				"path":   "/docs",
+			}},
+		}}
+
+		pages := expandPages(input)
+		if pages == nil {
+			t.Fatal("pages is nil")
+		}
+		if pages.GetBuildType() != "legacy" {
+			t.Errorf("got %q; want %q", pages.GetBuildType(), "legacy")
+		}
+		if pages.GetSource().GetBranch() != "main" {
+			t.Errorf("got %q; want %q", pages.GetSource().GetBranch(), "main")
+		}
+		if pages.GetSource().GetPath() != "/docs" {
+			t.Errorf("got %q; want %q", pages.GetSource().GetPath(), "/docs")
+		}
+	})
 }
 
 func TestAccGithubRepositorySecurity(t *testing.T) {

